core/types/vote.go

```diff
@@ -0,0 +1,89 @@
+package types
+
+import (
+	"sync/atomic"
+
+	"github.com/ethereum/go-ethereum/crypto/bls"
+	"github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	BLSPublicKeyLength = 48
+	BLSSignatureLength = 96
+
+	MaxAttestationExtraLength = 256
+)
+
+type BLSPublicKey [BLSPublicKeyLength]byte
+type BLSSignature [BLSSignatureLength]byte
+type ValidatorsBitSet uint64
+
+// VoteData represents the vote range that validator voted for fast finality.
+type VoteData struct {
+	TargetNumber uint64      // The target block number which validator wants to vote for.
+	TargetHash   common.Hash // The block hash of the target block.
+}
+
+// Hash returns the hash of the vote data.
+func (d *VoteData) Hash() common.Hash { return rlpHash(d) }
+
+// VoteEnvelope represents the vote of a single validator.
+type VoteEnvelope struct {
+	VoteAddress BLSPublicKey // The BLS public key of the validator.
```

Comment:
I think this should be renamed to PublicKey, address is not a terminology in BLS signature scheme.

Reply:
I think the vote is using bls public key, therefore, remove the `BLS` would make confusion

Reply:
I mean we should change the name of the variable from VoteAddress to PublicKey